rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -3270,18 +3267,36 @@ static rmw_ret_t rmw_take_ser_int(
 
       // taking a serialized msg from shared memory
 #ifdef DDS_HAS_SHM
-      if (sub->is_loaning_available && d->iox_chunk != nullptr) {
-        if (rmw_serialize(
-            SHIFT_PAST_ICEORYX_HEADER(d->iox_chunk), &sub->type_supports,
-            serialized_message) != RMW_RET_OK)
-        {
-          RMW_SET_ERROR_MSG("Failed to serialize sample from loaned memory");
+      if (d->iox_chunk != nullptr) {
+        auto shm_data_state = shm_get_data_state(d->iox_chunk);
+        if (shm_data_state == IOX_CHUNK_CONTAINS_SERIALIZED_DATA) {
+          size_t size = ddsi_serdata_size(d);
+          if (rmw_serialized_message_resize(serialized_message, size) != RMW_RET_OK) {
```

Comment:
Correct, but I have now realized that I can use `ddsi_sertype_get_serialized_size()` to get the serialized size here instead of  `ddsi_serdata_size()`, since its slightly better in performance than the latter as it will not sterilize the entire sample.

I will change this.